### PR TITLE
Add ability to disable an entity

### PR DESCRIPTION
**Pull request by jefferai**
This disallows its associated tokens to be used, but doesn't revoke
them.

### CODE REVIEW

File: helper/identity/types.proto [hunk 1]
```diff
@@ -110,6 +110,10 @@ message Entity {
 	// MFASecrets holds the MFA secrets indexed by the identifier of the MFA
 	// method configuration.
 	//map<string, mfa.Secret> mfa_secrets = 10;
```
Comment (briankassouf) [inline helper/identity/types.proto]: I’m suspicious having this commented out might break if a user upgrades from oss -> ent

File: vault/core.go [hunk 2]
```diff
@@ -1466,6 +1477,12 @@ func (c *Core) StepDown(req *logical.Request) (retErr error) {
 		return retErr
 	}
 
+	if entity != nil && entity.Disabled {
```
Comment (vishalnayak) [inline vault/core.go]: I wonder if we should do this before audit logging.

File: vault/core.go [hunk 3]
```diff
@@ -1327,12 +1338,6 @@ func (c *Core) sealInitCommon(ctx context.Context, req *logical.Request) (retErr
 		// for validation and the operation should be performed. But for now,
```
Comment (vishalnayak) [inline vault/core.go]: Can we remove these comments?

Comment (jefferai) [reply vault/core.go]: Oh, I should move them down.

File: vault/core.go (hunk 2 above)
Comment (jefferai) [reply vault/core.go]: No -- for the same reason that we attempt to log whatever we get back from the token store before fully checking validity. The fact that a request is invalid doesn't mean it wasn't an attempted request.

**Pull request merged**